### PR TITLE
Update spack-stack build instructions for Intel based Mac about handling pkg-config

### PR DESCRIPTION
**Pull request by srherbener**
### Summary

Changes handling of pkg-config on Intel based Mac to a special case. Note this PR supercedes PR #688. We decided to only homebrew install pkg-config for the Intel based Mac

### Testing

Successfully built spack-stack on Mac with pkg-config installed by homebrew. 

### Applications affected

None - purely a documentation update (no code changes)

### Systems affected

Intel based Mac

### Dependencies

None

### Issue(s) addressed

None

### Checklist
- [x] This PR addresses one issue/problem/enhancement, or has a very good reason for not doing so (special case for getting spack-stack built on Intel based Mac)
- [x] These changes have been tested on the affected systems and applications.
- [ ] All dependency PRs/issues have been resolved and this PR can be merged.

### CODE REVIEW

**Review by ericlingerfelt — APPROVED** [review]
Thanks, @srherbener!

File: doc/source/NewSiteConfigs.rst [hunk 1]
```diff
@@ -143,6 +143,12 @@ These instructions are meant to be a reference that users can follow to set up t
    brew install qt@5
    brew install mysql
 
+.. note::
+  On an Intel based Mac, you will need to also install pkg-config using homebrew.
+  This is done to work around an issue where libraries (eg, openssl) cannot be properly found during code compilation.
+.. code-block:: console
```
Comment (climbfuji) [inline doc/source/NewSiteConfigs.rst:150]: This isn't rendered correctly (not showing up: https://spack-stack--726.org.readthedocs.build/en/726/NewSiteConfigs.html#macos)

File: doc/source/NewSiteConfigs.rst [hunk 2]
```diff
@@ -220,6 +226,12 @@ Remember to activate the ``lua`` module environment and have MacTeX in your sear
    # The texlive bin directory must have been added to PATH (see above)
    spack external find --scope system texlive
 
+.. note::
+  On an Intel based Mac, you need to add the following spack config command to prevent spack from building pkg-config.
+  This will force spack to use the pkg-config installed by homebrew (see above).
+.. code-block:: console
```
Comment (climbfuji) [inline doc/source/NewSiteConfigs.rst:235]: Same here. Definitely blank lines are missing

**Review by climbfuji — CHANGES_REQUESTED** [review]

**Review by climbfuji — APPROVED** [review]
Documentation looks good now, thanks

**Pull request merged**